app/services/charges/validators/graduated_service.rb

```diff
@@ -30,9 +28,14 @@ def ranges
         charge.properties.map(&:with_indifferent_access)
       end
 
-      def valid_amounts?(range)
-        ::Validators::DecimalAmountService.new(range[:per_unit_amount]).valid_amount? &&
-          ::Validators::DecimalAmountService.new(range[:flat_amount]).valid_amount?
+      def validate_amounts(range)
+        unless ::Validators::DecimalAmountService.new(range[:per_unit_amount]).valid_amount?
+          add_error(field: :per_unit_amount, error_code: 'invalid_amount')
+        end
+
+        unless ::Validators::DecimalAmountService.new(range[:flat_amount]).valid_amount?
```

Comment:
Style/GuardClause: Use a guard clause (`return if ::Validators::DecimalAmountService.new(range[:flat_amount]).valid_amount?`) instead of wrapping the code inside a conditional expression. (https://rubystyle.guide#no-nested-conditionals)